### PR TITLE
X modal

### PR DESCRIPTION
**Pull request by tbowenbw**
## PR Checklist

Check these before submitting your pull request:

- [X] Visual and behavioral components are separated
- [X] Exported components are documented with examples
- [X] Props have JSDoc comments
- [?] All relevant visual sub-components can be overridden via props

## Breaking Changes

* `onBlockerClicked` prop removed from `Modal` and replaced with `onClose` instead, which is fired when either the user clicks on the `ModalBlocker` or when the user clicks on the `Anchor` embedded in the default `ModalTitle`

## Changes to Existing Components

* Adding an X to existing Modal

## New Visual Components

None
## New Behavioral Components

None

### CODE REVIEW

**Comment by tbowenbw** [comment]
Unsure if inserting an `Anchor` into the `Title` lets the developer override all sub-components.  Perhaps `ModalTitle` should be a proper `React.Component` which itself uses the `H3` and `Anchor` so that the developer can override this with another component if desired.  Interested specifically in feedback around this topic.

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -18,9 +19,9 @@ class Modal extends React.Component {
      */
     title: PropTypes.string,
     /**
-     * Handles click events on the backdrop behind the modal. Use this to close it if desired.
+     * Handles click events on the backdrop behind the modal or on the X in the title bar.
      */
-    onBlockerClicked: PropTypes.func,
+    onClose: PropTypes.func,
```
Comment (zelitzsi) [inline src/components/Modal/Modal.js]: We should mark `onClose` as required.

Comment (a-type) [reply src/components/Modal/Modal.js]: Some modals are closed by internal button controls, which would force the user to pass a no-op to the modal component itself.

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -77,25 +78,34 @@ class Modal extends React.Component {
     event.stopPropagation();
   };
 
+  renderTitle = () => {
+    const { title, onClose, Title } = this.props;
+    if (!title) return null;
+    return (
+      <Title>
+        {title}
+        <Anchor icon="delete2" onClick={onClose} />
```
Comment (a-type) [inline src/components/Modal/Modal.js]: Adding a truthiness check for `onClose` to conditionally render the anchor would let us not render a close button that doesn't do anything.

File: src/components/Modal/Modal.js [hunk 3]
```diff
@@ -77,25 +78,34 @@ class Modal extends React.Component {
...
       Blocker,
       Content,
       Window,
       ActionContent,
-      Title,
     } = this.props;
 
     return (
-      <Blocker onClick={onBlockerClicked}>
+      <Blocker onClick={onClose}>
```
Comment (a-type) [inline src/components/Modal/Modal.js]: Please fallback to `onBlockerClicked` to keep compatibility.

File: src/components/Modal/Modal.js (hunk 2 above)
Comment (a-type) [reply src/components/Modal/Modal.js]: Also, to adhere to the 'overrideable props' rule, you'll want to make `CloseButton` a prop for this component and use it here, and provide the SCL anchor as a default prop.

File: src/components/Modal/styles/ModalTitle.js [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import SectionTitle from 'components/SectionTitle';
 import styled from 'styled-components';
+import Anchor from 'components/Anchor';
 
 export default styled(SectionTitle)`
   padding: 0.95em 1em 0.95em 1.5em;
+
+  & > a {
```
Comment (a-type) [inline src/components/Modal/styles/ModalTitle.js]: A stronger selector would be to pass a class name to the close button anchor and reference it here instead. This works hand-in-hand with the overrideable prop suggestion.

**Pull request merged**